container_checkpoint_opts.go

```diff
@@ -98,8 +98,8 @@ func WithCheckpointTask(ctx context.Context, client *Client, c *containers.Conta
 
 // WithCheckpointRuntime includes the container runtime info
 func WithCheckpointRuntime(ctx context.Context, client *Client, c *containers.Container, index *imagespec.Index, copts *options.CheckpointOptions) error {
-	if c.Runtime.Options != nil {
-		data, err := c.Runtime.Options.Marshal()
+	if c.Runtime.Options != nil && c.Runtime.Options.GetValue() != nil {
```

Comment:
This is too error-prone. I actually want to have https://github.com/containerd/typeurl/pull/33 instead and check only GetValue().

Reply:
The PR has been merged, but if Options is really nil, having GetValue on the typed nil pointer wouldn't help... So we need to live with this.